### ISSUE TITLE
Link from App Router to Page Router removes trailing 4 characters

### ISSUE THREAD

**Issue by AndrewLeedham** [vercel/next.js#51845]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:22 PDT 2023; root:xnu-8796.121.3~7/RELEASE_X86_64
    Binaries:
      Node: 16.18.0
      npm: 9.6.3
      Yarn: 1.22.19
      pnpm: 8.6.1
    Relevant packages:
      next: 13.4.7
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.3
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue or a replay of the bug

https://stackblitz.com/edit/stackblitz-starters-vpt9gp?file=app%2Fpage.js

### To Reproduce

I have set the stackblitz up to build and start a local static server. Clicking the about link once loaded should navigate to `/a` and 404 because it does not exists. I have tried this with various links and it always removes the last 4 characters.

I also tried this with the latest canary: https://stackblitz.com/edit/stackblitz-starters-ewpg3m?file=package.json same issue.

### Describe the Bug

Linking from a partially migrated App Router to a page from the old Page Router 404s as it tries to navigate to the url with the last 4 characters missing. The `href` URL in the browser is correct, but the JavaScript "Soft navigation" is failing.

### Expected Behavior

Links from App Router to Page Router pages to work while migrating between the 2.

### How are you deploying your application? (if relevant)

next build

**Comment by icyJoseph** [comment]
Wow, but only when doing export mode. What an intriguing bug.

**Comment by scriptedcity** [comment]
I faced the similar issue but I'm only using App router.

In my case, adding attribute `prefetch={false}` would prevent this.

**Comment by icyJoseph** [comment]
Aha, so it is failing to prefetch its own RSC data format 🤔  - I don't have enough knowledge of how the App Router, router, is implemented, I couldn't find the `window.location.href =`, or equivalent statement, that's changing the location to `/a` -

**Comment by harryli0088** [comment]
I'm getting the same problem for an app I'm trying to deploy to GitHub pages, even with `pretch={false}`. If you click on the go home button, it has the same bug that removes the 4 trailing characters. But if you inspect the anchor element's href, it is correct. It even redirects properly if you click on the href link in the inspector.

https://harryli0088.github.io/nextjs-github-pages/test-route

**Comment by idrahoner** [comment]
Hello
Faced with such a problem [next 13.4.10 App Router]. It kept removing the last 4 letters of the home link. Adding property `{trailingSlash: true}` to the `next.config.js` helped. It worked on two repositories.
Link to the example:

- GH Pages: https://idrahoner.github.io/oneword/another/
- Repository: https://github.com/idrahoner/oneword

**Comment by icyJoseph** [comment]
Alright, this has been fixed in `next@13.4.11`:

https://github.com/vercel/next.js/releases

**Comment by AndrewLeedham** [comment]
Working beautifully. Thanks for tracking it down @icyJoseph :)
